plone/protect/utils.py

```diff
@@ -122,5 +124,11 @@ def safeWrite(obj, request=None):
     try:
         if obj._p_oid not in request.environ[SAFE_WRITE_KEY]:
             request.environ[SAFE_WRITE_KEY].append(obj._p_oid)
+        if isinstance(obj, (OOBTree, IOBTree)):
```

Comment:
There are more BTree-Types around than those two. But all are inheriting from https://github.com/zopefoundation/BTrees/blob/master/BTrees/_base.py#L775 where also `_firstbucket` is introduced.
because we are relying here anyway on internal structures I propose to check against this "private" `_Tree` instead against all variants.

Reply:
Great, thanks for the hint!

Reply:
Meh, it looks like they don't properly inherit after all...
```
>>> obj.__class__.__mro__
(<class 'BTrees.OOBTree.OOBTree'>, <class 'persistent.Persistent'>, <class 'object'>)
```

Reply:
:man_facepalming: probably b/c of the C-implementation or something like this....

Reply:
Quite probable. So what do we do? I wouldn't check for the attribute because some other class might have it. As it's a security thing I'd rather add fewer exceptions than more. So I'd propose to leave it as it is and only add more types to the list if and when we find a practical need for it.

Reply:
I think we only have these 8 trees. I dont think there will be a new one in near future. We should support them all from the beginning, as the bucket problem is always the same for all.
- IFBTree
- IIBTree
- IOBTree
- LFBTree
- LLBTree
- LOBTree
- OIBTree
- OLBTree

Reply:
The OOBTree seems to be missing in the list. Is it 9, then?

Reply:
Yes, 9 seems to be right.

Reply:
oh, yes!